pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,411 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var cockpit = require("cockpit");
+var _ = cockpit.gettext;
+
+var client = { };
+
+cockpit.event_target(client);
+
+client.subscriptionStatus = {
+    serviceStatus: undefined,
+    status: undefined,
+    products: [],
+    error: undefined,
+};
+
+// we trigger an event called "dataChanged" when the data has changed
+
+// DBUS service
+var service;
+
+function needRender() {
+    var ev = document.createEvent("Event");
+    ev.initEvent("dataChanged", false, false);
+    client.dispatchEvent(ev);
+}
+
+/* we trigger status update via dbus
+ * if we don't get a timely reply, consider subscription-manager failure
+ */
+var updateTimeout;
+
+/*
+ * Parses lines like:
+ *
+ * id:  text
+ */
+function parseSingleSubscription(text) {
+    var ret = { };
+    text.split('\n').forEach(function(line, i) {
+        var pos = line.indexOf(':');
+        if (pos !== -1)
+            ret[line.substring(0, pos)] = line.substring(pos + 1).trim();
```

Comment:
should work as is, but it doesn't hurt to add an extra trim in case the whitespace changes